src/base/static/js/views/place-detail-view.js

```diff
@@ -62,42 +143,315 @@
+            model: self.model,
+            placeDetailView: self,
+            fieldName: $(this).attr("name"),
+            fieldId: $(this).attr("id")
+          });
+        });
+      }
 
       return this;
     },
+
+    tearDown: function() {
+      this.options.router.off("route", this.tearDown, this);
+      this.isEditingToggled = false;
+      this.isModified = false;
+    },
+
+    onModified: function() {
+      this.isModified = true;
+      $("#update-place-model-btn").addClass("isModified");
+      $(this.watchFields).off("keyup change");
+    },
+
     remove: function() {
       // Nothing yet
     },
 
     onChange: function() {
       this.render();
+    },
+
+    // called by the router
+    onCloseWithUnsavedChanges: function() {
+      // if (confirm("You have unsaved changes. Proceed?")) {
+      //   this.isModified = false;
+      //   return true;
+      // }
+
+      // return false;
+      
+      return true;
+    },
+
+    onBinaryToggle: function(evt) {
+      var oldValue = $(evt.target).val(),
+          newValue = $(evt.target).data("alt-value"),
+          oldLabel = $(evt.target).siblings("label").html(),
+          newLabel = $(evt.target).siblings("label").data("alt-label");
+
+      // swap new and old values and labels
+      $(evt.target).data("alt-value", oldValue);
+      $(evt.target).val(newValue);
+      $(evt.target).siblings("label").html(newLabel);
+      $(evt.target).siblings("label").data("alt-label", oldLabel);
+    },
+
+    scrapeForm: function() {
+      var self = this,
+      richTextAttrs = {};
+
+      // attach data from Quill-enabled fields
+      this.$el.find(".ql-editor").each(function() {
+        richTextAttrs[$(this).data("fieldName")] = $(this).html();
+      });
+
+      this.$el.find("input[name='url-title']").each(function() {
+        $(this).val(Util.prepareCustomUrl($(this).val()));
+      });
+
+      var attrs = _.extend(Util.getAttrs($("#update-place-model-form")), 
+        Util.getAttrs($("#update-place-model-title-form")),
+        richTextAttrs);
+
+      // Special handling for binary toggle buttons: we need to remove
+      // them completely from the model if they've been unselected in
+      // the editor
+      this.$el.find("#update-place-model-form input[data-input-type='binary_toggle']").each(function(input) {
+        if (!$(this).is(":checked")) {
+          self.model.unset($(this).attr("id"));
+        }
+      });
+
+      // Special handling for checkbox groups: if all items in a checkbox group
+      // have been unselected, remove the group completely from the model
+      this.$el.find("#update-place-model-form .checkbox-group").each(function(group) {
+        if ($(this).find("input:checkbox:checked").length === 0) {
+          self.model.unset($(this).find(":first-child").attr("name"));
+        }
+      });
+      
+      return attrs;
+    },
+
+    // TODO: can we relax any of the validation rules in edit mode?
```

Comment:
yes-- I think we need all the same validation rules here as we have in the form.